### PR TITLE
Add `rehype-sectionize` to list of plugins

### PR DESCRIPTION
**Pull request by hbsnow**
### Initial checklist

* [x] I read the support docs 
* [x] I read the contributing guide 
* [x] I agree to follow the code of conduct 
* [x] I searched issues and couldn’t find anything (or linked relevant results below) 
* [x] If applicable, I’ve added docs and tests

### Description of changes

The rehype-section was not maintained.
Therefore, this plugin was created to add the id of the heading to the section.

### CODE REVIEW

**Comment by ChristianMurphy** [comment]
Welcome @hbsnow!
Thanks for sharing! 🙇 

I'm not sure about removing the existing plugin 🤔 
It hasn't been updated for a bit, but I don't see a deprecation notice on it.
I might recommend keeping both for now.
Thoughts @rehypejs/maintainers?

---

Some pointers and suggestions for the package.
* Consider building the ESM version with `"module": "node16"`, to ensure it will be compatible with native ESM (https://www.typescriptlang.org/tsconfig#module)
* Unless you have a very specific CJS requirement, it would probably be best to export only the ESM version, and allow downstream tools decide if/when/how they want to covert things to another format.
* When `type: module` is set in package.json, `main` should be ESM. (currently you have it set to CJS which will likely break Node js users)
* In _package.json_ for `dependencies` use a wide `^` range where possible. Pinning specific versions of dependencies can cause version conflicts and duplication in the _node_modules_ folder.
* Node natively provides `test` (https://nodejs.org/api/test.html), `assert` (https://nodejs.org/api/assert.html), and coverage. Vite and Vitetest aren't needed and may add some complexity to the setup

**Comment by remcohaszing** [comment]
I agree, let’s add the new one, but also keep the old one.

I also agree with all if the tips given, but I would be cautious to recommend `node:test`. It’s still an experimental feature and it’s available as of Node 16, but Node 14 is still supported.

Additionally I always recommend to add `lockfile-version = 3` to `.npmrc`.

**Review by charlestonXdao — APPROVED** [review]
good

**Comment by hbsnow** [comment]
@ChristianMurphy @remcohaszing 
Thanks a lot for the advice, I fixed the library code.

I have restored the existing plugin.

File: doc/plugins.md [hunk 1]
```diff
@@ -166,6 +166,8 @@ The list of plugins:
     — sanitize HTML
 *   [`rehype-section`](https://github.com/agentofuser/rehype-section)
     — wrap headings and their contents into nested `<section>` elements
+*   [`rehype-sectionize`](https://github.com/hbsnow/rehype-sectionize)
+    — when heading has id, section with id wraps the heading and content
```
Comment (wooorm) [inline doc/plugins.md:170]: I don’t understand this text. Perhaps something like the following, based on the description of `rehype-section`.

```suggestion
    — wrap headings (with IDs) and their contents into nested `<section>`
    elements, with attributes
```

But like, what does your plugin do when there are no IDs on headings? Maybe you don’t need to include the `(with IDs)` part here.

Comment (hbsnow) [reply doc/plugins.md]: I'm not very good at English, thank you!

**Review by ChristianMurphy — APPROVED** [review]
Thanks @hbsnow!

LGTM, thanks for sharing with the community. 🙇 

A few more, completely optional ideas:
* in _tsconfig.json_ when `module: node16` is set, `moduleResolution` and `esModuleInterop` shouldn't be needed https://github.com/hbsnow/rehype-sectionize/blob/8ec81ef75e6e3290040d3cf1f2cc878b0c9f0768/tsconfig.json#L5-L6
* It could be good to note a bit more on different ways it can be installed (if it supports deno/browser as well, which it looks like it may) as well as how it can be used programmatically with a little code snippet (see https://github.com/rehypejs/rehype-slug#install for some inspiration)
* For dependency ranges, it could/should be save to widen the version range to the first release in a major series, for example instead of `^2.1.1` one could say `^2.0.0` to allow package managers to de-duplicate if there is already a lower minor/patch version. https://github.com/hbsnow/rehype-sectionize/blob/8ec81ef75e6e3290040d3cf1f2cc878b0c9f0768/package.json#L46-L50
* TypeScript has a different/looser interpretation of SemVer, they can introduce breaking changes for libraries in minor releases. Depending on how you want to handle that, you could use a `~` to lock down a specific minor release in tests, or keep it `^` with the expectation that it may break every now and again https://github.com/hbsnow/rehype-sectionize/blob/8ec81ef75e6e3290040d3cf1f2cc878b0c9f0768/package.json#L43
* it looks like it could be safe to add `sideEffects: false` to _package.json_ which helps bundlers like `webpack` with tree shaking (https://webpack.js.org/guides/tree-shaking/#mark-the-file-as-side-effect-free)
* Unless you specifically need a script to run during the install phase (I don't see any currently) I'd recommend setting `ignore-scripts=true` in _.npmrc_ to limit supply chain risks from upstream packages having unnecessary life cycle scripts.

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
Thank you! :)

**Pull request merged**